include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -110,7 +110,7 @@ inline constexpr __is_onedpl_device_accessible_content_iterator_fn __is_onedpl_d
 
 template <typename T>
 struct is_device_accessible_content_iterator
-    : decltype(oneapi::dpl::__internal::__is_onedpl_device_accessible_content_iterator(std::declval<T>())){};
+    : std::bool_constant<bool(decltype(oneapi::dpl::__internal::__is_onedpl_device_accessible_content_iterator(std::declval<T>()))::value)>{};
```

Comment:
Am I correct that the result of `oneapi::dpl::__internal::__is_onedpl_device_accessible_content_iterator(std::declval<T>())` is the value of `true_type` or `false_type` ?
If so I don't understood why we need conversion to `bool` type here and then `std::bool_constant`...

Reply:
OK, it was just my question, I am not going to argue with this approach.